### PR TITLE
Update eventName=Screen to $screen to match PostHog's special behaviour

### PR DESCRIPTION
**Pull request by t3chguy**

### CODE REVIEW

File: types/swift/$screen.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+//
+
+import Foundation
+
+// GENERATED FILE, DO NOT EDIT. FOR MORE INFORMATION VISIT
+// https://github.com/matrix-org/matrix-analytics-events/
+
+/// Triggered when the user changed screen
+extension AnalyticsEvent {
+    public struct $screen: AnalyticsEventProtocol {
```
Comment (pixlwave) [inline types/swift/$screen.swift:24]: @t3chguy This wouldn't be a valid class name in Swift. Could we leave the filename as `Screen.json` and only update the `eventName` property?

Comment (t3chguy) [reply types/swift/$screen.swift:24]: This wouldn't match the guidelines set out by this repo

https://github.com/matrix-org/matrix-analytics-events#guidelines-for-creating-new-event-schema

> The filename should match the event's name, including case.

But yes it is an exception we will have to make.

Comment (pixlwave) [reply types/swift/$screen.swift:24]: Ah that's annoying to have guidelines and then go an break them 😕
Maybe we should update that sentence to mention dropping special characters, and the if necessary on the mobile side we could update the generator to always upper case the first letter of the class name.

**Review by pixlwave — APPROVED** [review]
Works for me, thanks 🙏.

**Pull request merged**